core/runtime/src/main/java/io/quarkus/runtime/QuarkusApplication.java

```diff
@@ -1,6 +1,17 @@
 package io.quarkus.runtime;
 
+/**
+ * Defines a class as a QuarkusApplication ran by {@link Quarkus}. This is usually used for command mode applications
+ * with startup logic. The logic is executed inside {@link QuarkusApplication#run} method before
+ * the main application exits.
+ */
 public interface QuarkusApplication {
 
+    /**
+     * Method to run a QuarkusApplication.
```

Comment:
I don't think this method really needs any Javadoc to be honest

Reply:
For now let's leave it undocumented. Maybe in the future some new customers (framework users) will ask what's the usage of this run function in a method level for their convenience.